pallets/automation-price/src/lib.rs

```diff
@@ -995,24 +1008,78 @@ pub mod pallet {
 			Ok(now)
 		}
 
-		/// Runs as many tasks as the weight allows from the provided vec of task_ids.
-		///
-		/// Returns a vec with the tasks that were not run and the remaining weight.
-		pub fn run_tasks(
-			mut task_ids: TaskIdList<T>,
-			mut weight_left: Weight,
-		) -> (TaskIdList<T>, Weight) {
-			let mut consumed_task_index: usize = 0;
+		// Check whether a task can run or not based on its expiration and price.
+		//
+		// A task can be queued but got expired when it's about to run, in that case, we don't want
```

Comment:
`A task can be queued but got expired when it's about to run, in that case, we don't want it to be run.`
While there may be different opinions on whether it should be executed or not, we can keep the behavior as-is. This comment is crucial for reviewing the design decision later. 👌